src/selectors/entities/teams.js

```diff
@@ -136,14 +137,75 @@ export function getTeamMember(state, teamId, userId) {
     return null;
 }
 
+export const getListableTeamIds = createIdsSelector(
+    getTeams,
+    getTeamMemberships,
+    (state) => haveISystemPermission(state, {permission: Permissions.LIST_PUBLIC_TEAMS}),
+    (state) => haveISystemPermission(state, {permission: Permissions.LIST_PRIVATE_TEAMS}),
+    (state) => state.entities.general.serverVersion,
+    (teams, myMembers, canListPublicTeams, canListPrivateTeams, serverVersion) => {
+        return Object.keys(teams).filter((id) => {
+            const team = teams[id];
+            const member = myMembers[id];
+            let canList = team.allow_open_invite;
+            if (isMinimumServerVersion(serverVersion, 5, 10, 0) ||
+                   (serverVersion.indexOf('dev') !== -1 && isMinimumServerVersion(serverVersion, 5, 8, 0)) ||
+                   (serverVersion.match(/^5.8.\d.\d\d\d\d.*$/) !== null && isMinimumServerVersion(serverVersion, 5, 8, 0))) {
+                canList = (canListPrivateTeams && !team.allow_open_invite) || (canListPublicTeams && team.allow_open_invite);
+            }
+            return team.delete_at === 0 && canList && !member;
```

Comment:
Here I'm filtering by `!member`, but I'm not 5/5 about it. Maybe the name of the selector should be something like `getListableNotMemberTeamIds` and propagate this name for the other 2 functions.

Reply:
Makes sense to me.